addons/website/tests/test_website_reset_password.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+
+import odoo
+from odoo.tests import tagged
+from odoo.tests.common import HttpCase
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteResetPassword(HttpCase):
+
+    def test_01_website_reset_password_tour(self):
+        """The goal of this test is to make sure the reset password works."""
+
+        # we override unlink because we don't want the email to be auto deleted
+        MailMail = odoo.addons.mail.models.mail_mail.MailMail
+
+        # we override send_mail because in HttpCase we don't have an smtp server
+        # without force_send and raise_exception, it is going to fail silently
+        MailTemplate = odoo.addons.mail.models.mail_template.MailTemplate
+        original_send_mail = MailTemplate.send_mail
+
+        def my_send_mail(*args, **kwargs):
+            kwargs.update(force_send=False, raise_exception=False)
+            return original_send_mail(*args, **kwargs)
+
+        with patch.object(MailMail, 'unlink', lambda self: None), patch.object(MailTemplate, 'send_mail', my_send_mail):
```

Comment:
??

why not simply send the email (that is btw catched by runbot in a mailctacher? )?

Reply:
ok, I know your futur answer, yes it is sent...

but why force_send=False, raise_exception=False is needed ?

Reply:
ok, I know you futur answer, yes it is explain

# we override send_mail because in HttpCase we don't have an smtp server
# without force_send and raise_exception, it is going to fail silently

But what is the problem to have a mail that fail?

Reply:
What I want is:
- if the send works, I don't want the mail to be deleted
- if the send fails, I don't want an exception to be raised and I don't want an error in the log

When `force_send` is false, `raise_exception` is not read so I could remove that second arg. But the rest is necessary for the 2 cases above.